txpool/txpool.go

```diff
@@ -471,60 +468,164 @@ func (t *TxPool) ResetWithHeader(h *types.Header) {
 	evnt := &blockchain.Event{
 		NewChain: []*types.Header{h},
 	}
+
+	// Process the txns in the event to make sure the TxPool is up-to-date
 	t.ProcessEvent(evnt)
 }
 
-// ProcessEvent processes the blockchain event and resets the txpool accordingly
+// ProcessEvent processes the blockchain event and updates the txpool accordingly
 func (t *TxPool) ProcessEvent(evnt *blockchain.Event) {
```

Comment:
Please note that you need to recover transactions from old block in case of other consensuses. You need to revert some transactions if reorg happens. but we don't have such consensus right now

Reply:
Exactly 👍 

Do you feel like the removed section for the `OldChain` is something that we should leave intact?

@lazartravica @munna0908 curious to hear your thoughts on this as well

Reply:
@zivkovicmilos let's not alter any more than we need to right now, we will tackle this when refactoring next week.

Reply:
@zivkovicmilos I do agree with youshiki, if we're planning to support various consensus down the line it make sense to have the oldchain logic intact

Reply:
Got it 👍 

Reverted that change here:

bd95c19